Detectors/PHOS/base/include/PHOSBase/Digit.h

```diff
@@ -0,0 +1,137 @@
+  /// \return constructed Digit
+  Digit(Hit hit);
+
+  ~Digit() = default; // override
+
+  /// \brief Replace content of this digit with new one, from hit
+  /// \param PHOS Hit
+  /// \return
+  void FillFromHit(Hit hit);
+
+  /// \brief Comparison oparator, based on time and absId
+  /// \param another PHOS Digit
+  /// \return result of comparison: first time, if time same, then absId
+  bool operator<(const Digit& other) const;
+  /// \brief Comparison oparator, based on time and absId
+  /// \param another PHOS Digit
+  /// \return result of comparison: first time, if time same, then absId
+  bool operator>(const Digit& other) const;
+  /// \brief Check, if one can add two digits
+  /// \param another PHOS Digit
+  /// \return true if time stamps are same and absId are same
+  bool canAdd(const Digit other) const;
+  /// \brief if addable, adds energy and list of primaries.
+  /// \param another PHOS Digit
+  /// \return digit with sum of energies and longer list of primaries
+  Digit& operator+=(const Digit& other); //
+
+  /// \brief Absolute sell id
+  Int_t getAbsId() const { return mAbsId; }
+  void setAbsId(Int_t cellId) { mAbsId = cellId; }
+
+  /// \brief Energy deposited in a cell
+  Double_t getAmplitude() const { return mAmplitude; }
+  void setAmplitude(Double_t amplitude) { mAmplitude = amplitude; }
+
+  /// \brief time measured in digit w.r.t. photon to PHOS arrival
+  Double_t getTime() const { return mTime; }
+  void setTime(Double_t time) { mTime = time; }
+
+  /// \brief Checks if this digit is produced in High Gain or Low Gain channels
+  Double_t isHighGain() const { return mIsHighGain; }
+  void setHighGain(Bool_t isHG) { mIsHighGain = isHG; }
+
+  /// \brief Label of a particle made energy deposition
+  /// \param idx index in a list of a particles, max length kMaxLabels
+  /// \return lable of a particle. Lables are sorted according to energy deposited by each of them
+  Label getLabel(Int_t idx) const
+  {
+    if (idx < kMaxLabels)
+      return mLabels[idx];
+    else
+      return -1;
+  }
+  /// \brief Proportion of energy deposited by particle idx
+  /// \param idx index in a list of a particles, max length kMaxLabels
+  /// \return Proportion of energy from this particle.
+  Label getLabelEProp(Int_t idx) const
+  {
+    if (idx < kMaxLabels)
+      return mEProp[idx];
+    else
+      return 0.;
+  }
+  /// \brief Number of particles assosiated with this digit
+  int getNLabels() const { return mNlabels; }
+
+  void PrintStream(std::ostream& stream) const;
+
+ private:
+  // friend class boost::serialization::access;
+
+  int mAbsId;                ///< cell index (absolute cell ID)
+  double mAmplitude;         ///< Amplitude
+  double mTime;              ///< Time
+  int mNlabels;              ///< Number of actual labels in this digit
+  Label mLabels[kMaxLabels]; ///< Particle labels associated to this digit
+  double mEProp[kMaxLabels]; ///< Proportion of total energy deposited by given primary
```

Comment:
As discussed in the last WP12 meeting, the preferred way of keeping these associations is via the o2::MCTruthContainer class which will ensure minimal memory usage.
To keep both the MCLabel as well as the energy, you could **for instance** make a new label class `using Label = std::pair<o2::MCCompLabel, double>` and store the labels inside `o2::MCTruthContainer<Label> mLabels`.

(I don't consider this a blocker and you can do attack this later of if preferred).

Reply:
Both Digit and Cluster are transient objects. Probably there is no much sense in reducing their size. Anyway I will study what you suggest.

Reply:
Are you saying that you never write these objects to disc?
But even if they truly transient it doesn't hurt to reduce their memory occupation.

I agree that we can do it in a next step.

Reply:
No, they are transient in the sense they are not stored in ESD/AOD.